src/ansible_navigator/utils/serialize.py

```diff
@@ -26,6 +43,37 @@
 # pylint: enable=unused-import
 
 
+def serialize(
+    content_view: "ContentView",
+    content: "ContentBase",
+    serialization_format: SerializationFormat,
+    file_mode: str = "w",
+    filename: Optional[Union[Path, str]] = None,
+) -> Optional[str]:
+    """Serialize a dataclass based on format and view.
+
+    :param content_view: The contest view
+    :param content: The content dataclass to serialize
+    :param serialization_format: The serialization format
+    :param file_mode: The mode for the file operation
+    :param filename: A filename to write to
+    :returns: The serialized content
+    """
+    content_as_dict = content.asdict(
```

Comment:
While the public API is permissive and accepts both `str` and `Path`, the internals don't need to work with `str`. So make sure it's path-like right away:
```suggestion
    if filename is not None:
        filename = Path(filename)
    content_as_dict = content.asdict(
```
Then, in the converters, only work with `Path`.

Reply:
If we change the signature of human dump to Optional[Path] it will cause typing issues elsewhere in the code:

```
src/ansible_navigator/actions/open_file.py: note: In member "run" of class "Action":
src/ansible_navigator/actions/open_file.py:109:50: error: Argument "filename" to "human_dump" has incompatible type "str"; expected "Optional[Path]"  [arg-type]
```

That said, If we leave human_dump as Path or str, we can make serialize Optional[Path].  This will require the caller be updated to a Path before they migrate to serialize, if they write to a file, which is not all that common.

It just raises the priority of this a little: https://github.com/ansible/ansible-navigator/issues/697